src/main/java/io/codelex/java/advanced/test/Exercise1/CreditCard.java

```diff
@@ -0,0 +1,35 @@
+package io.codelex.java.advanced.test.Exercise1;
+
+import java.math.BigDecimal;
+
+public class CreditCard extends Card{
+    public CreditCard(String number, String ownerName, String ccv, BigDecimal balance) {
+        super(number, ownerName, ccv, balance);
+    }
+
+    @Override
+    public BigDecimal addFunds(BigDecimal funds) {
+        BigDecimal updatedBalance = super.addFunds(funds);
+        System.out.println("Remaining funds: " + updatedBalance);
+        return updatedBalance;
+    }
+
+    @Override
+    public BigDecimal takeFunds(BigDecimal funds) throws NotEnoughFundsException {
+        BigDecimal remainingFunds = getBalance().subtract(funds);
+        if (remainingFunds.compareTo(BigDecimal.valueOf(100)) < 0) {
```

Comment:
Ja šo IF noliktu metodes beigās, tad varētu pamanīt, ka līdz tam IFam sakrīt abas takeFunds metodes abām kartēm. Tā kā vari to kopīgo daļu pārnest uz abstrakto Card klasi un šeit paliktu kaut kas tāds:
```
super.takeFunds(funds);
if (getBalance().compareTo(BigDecimal.valueOf(100)) < 0) {
    System.out.println("Warning: Low funds");
}
```
Savukārt no debit card to metodi varētu ņemt ārā jo viņa strādātu 1 pret 1 kā parent klasē.

Reply:
Bija tāda doma, bet baigi aizkavējos šeit un uzreiz ķeros klāt pie nākamā uzdevuma, atstājot kā ir.